components/analytics-core/org.wso2.carbon.analytics.datasource.core.test/src/main/java/org/wso2/carbon/analytics/datasource/core/AnalyticsDataServiceTest.java

```diff
@@ -293,6 +317,7 @@ private void indexDataAddRetrieve(int tenantId, String tableName, int n) throws
         List<Record> records = this.generateIndexRecords(tenantId, tableName, n, 0);
         this.service.put(records);
         this.service.waitForIndexing(DEFAULT_WAIT_TIME);
+        Thread.sleep(1000);
```

Comment:
`Thread.sleep()` in unit tests is evil (see [this](https://ardalis.com/thread-sleep-in-tests-is-evil) & [this](http://laurentkempe.com/2012/11/20/Using-ThreadSleep-in-unit-test-A-good-idea/)). Please remove.